packages/devtools_app/lib/src/screens/memory/shared/heap/heap.dart

```diff
@@ -33,12 +35,52 @@ class AdaptedHeap {
   }
 }
 
-abstract class HeapClasses with Sealable {
-  Iterable<ClassStats> get classStatsList;
+abstract class HeapClasses<T extends ClassStats> with Sealable {
+  List<T> get classStatsList;
+}
+
+mixin Filterable<T extends ClassStats> on HeapClasses<T> {
```

Comment:
make this private since this filtering mixin is specific to the memory code and shouldn't be used in other places in Devtools

Reply:
It is used outside the library. Renamed to FilterableHeapClasses to imply it is memory specific.